File: Solutions/283/csharp/Program.cs
using System;
using System.Collections.Generic;
using System.Linq;

namespace Quera {
    public static class Program {
        public static void Main() {
            var inputs = GetInputs<int>(2);

            PrintSquare(inputs[0], inputs[1]);
        }

        private static void PrintSquare(int outerSquare, int innerSquare) {
            if (innerSquare >= outerSquare) {
                Console.WriteLine("Wrong order!");
                return;
            }

            var diff = outerSquare - innerSquare;
            if (diff % 2 != 0) {
                Console.WriteLine("Wrong difference!");
                return;
            }

            var line = string.Concat(Enumerable.Repeat("* ", outerSquare));
            var innerLineStars = string.Concat(Enumerable.Repeat("* ", diff / 2));
            var innerLine = innerLineStars + string.Concat(Enumerable.Repeat("  ", innerSquare)) + innerLineStars;

            for (var i = 0; i < diff / 2; i++)
                Console.WriteLine(line);
            for (var i = 0; i < innerSquare; i++)
                Console.WriteLine(innerLine);
            for (var i = 0; i < diff / 2; i++)
                Console.WriteLine(line);
        }

        private static List<T> GetInputs<T>(int count) {
            var result = new List<T>(count);
            for (var i = 0; i < count; i++) {
                result.Add((T) Convert.ChangeType(Console.ReadLine(), typeof(T)));
            }

            return result;
        }
    }
}

File: Solutions/283/c++/program.cpp
// 2018-09-13

#include <iostream>

using namespace std;

//Define:

//Prototypes:
void printSquare(int, int);


int main()
{
	int biggerSquare, smallerSquare;
	cin >> biggerSquare >> smallerSquare;

	printSquare(biggerSquare, smallerSquare);

	return 0;
}


void printSquare(int biggerSquare, int smallerSquare)
{
	if (biggerSquare < 1 || smallerSquare < 1)
		throw invalid_argument("Argument is not valid.");

	if (smallerSquare >= biggerSquare)
	{
		cout << "Wrong order!" << endl;
		return;
	}

	if ((biggerSquare - smallerSquare) % 2 != 0)
	{
		cout << "Wrong difference!" << endl;
		return;
	}

	auto diff = (biggerSquare - smallerSquare) / 2;

	for (int r = 0; r < biggerSquare; r++)
	{
		for (int c = 0; c < biggerSquare; c++)
		{
			if (r >= diff && r < biggerSquare - diff && c >= diff && c < biggerSquare - diff)
			{
				cout << "  ";
			}
			else
			{
				cout << "* ";
			}
		}
		cout << endl;
	}
}